File: study/lua/src/app/scene/view/customactivity/CustomActivityWeekFundsV2Cell.lua
-- @Author panhoa
-- @Date 3.12.2019
-- @Role 

local ListViewCellBase = require("app.ui.ListViewCellBase")
local CustomActivityWeekFundsV2Cell = class("CustomActivityWeekFundsV2Cell", ListViewCellBase)
local CustomActivityConst = require("app.const.CustomActivityConst")


function CustomActivityWeekFundsV2Cell:ctor(rewardCallback)
    self._rewardCallback = rewardCallback

    local resource = {
        file = Path.getCSB("CustomActivityWeekFundsV2Cell", "customactivity"),
    }
    CustomActivityWeekFundsV2Cell.super.ctor(self, resource)
end

function CustomActivityWeekFundsV2Cell:onCreate()
    local size = self._resource:getContentSize()
    self:setContentSize(size.width, size.height)
end

-- @Role    Create Effect
function CustomActivityWeekFundsV2Cell:_createEffect(effect, state)
    -- body
    if effect == 0 then return end
    local selectedFlash = self["_nodeEffect"]:getChildByName("flash_effect")
    if selectedFlash == nil then
        local lightEffect = require("app.effect.EffectGfxNode").new("effect_chongzhi_guangyun6")
        lightEffect:setAnchorPoint(0, 0)
        lightEffect:play()
        --lightEffect:setScale(1.1)
        lightEffect:setVisible(state == 0)
        lightEffect:setName("flash_effect")
        self["_nodeEffect"]:addChild(lightEffect)
        lightEffect:setPosition(self["_nodeEffect"]:getContentSize().width* 0.5,
                                self["_nodeEffect"]:getContentSize().height * 0.5 + 8)
    else
        selectedFlash:setVisible(state == 0)
    end
end

-- @Role    UpdateUI
function CustomActivityWeekFundsV2Cell:updateUI(data)
    -- body
    local effect = tonumber(data.effects)
    local curDay = tonumber(data.day)
    --self._imageNormal:loadTexture(Path.getCustomActivityUI(data.reward_background1))
    --self._imageNormal:ignoreContentAdaptWithSize(true)
    self._imageDay:loadTexture(Path.getCustomActivityUI(CustomActivityConst.FUNDS_V2_DAY[curDay]))
    self._imageDay:ignoreContentAdaptWithSize(true)
    self._imageYuanBao:loadTexture(Path.getCustomActivityUI(CustomActivityConst.FUNDS_YUANBAO[effect+1]))
    self._imageYuanBao:ignoreContentAdaptWithSize(true)
    --self._imageSelected:loadTexture(Path.getCustomActivityUI(data.reward_effect))
    --self._imageSelected:ignoreContentAdaptWithSize(true)
    self._textNum:setString("x" ..tostring(data.reward_size_1))

    if data.isActived and data.canSignedDay then
        self:_createEffect(effect, data.canGet)
        self._imageShade:setVisible(data.canGet == 1)
        self._imageDuiGou:setVisible(data.canGet == 1)
        self._imageLarge:setVisible(data.canGet == 0)
        self._imageSelected:setVisible(data.canGet == 0)
        self._panelTouch:setVisible(data.canGet == 0)
        self._textState:setString(data.canGet == 1 and Lang.get("weekfunds_v2_state2") or Lang.get("weekfunds_v2_state3"))
        self._textState:setColor(data.canGet == 0 and Colors.FUNDSWEEK_V2_GOT or Colors.FUNDSWEEK_V2_NOTGOT)
    else
        self:_createEffect(effect, 0)
        self._imageShade:setVisible(false)
        self._imageDuiGou:setVisible(false)
        self._imageLarge:setVisible(false)
        self._imageSelected:setVisible(false)
        self._panelTouch:setVisible(false)
        self._textState:setString(Lang.get("weekfunds_v2_state1"))
        self._textState:setColor(Colors.FUNDSWEEK_V2_NOTGOT)
    end

    self._panelTouch:setEnabled(true)
    self._panelTouch:setSwallowTouches(false)
    self._panelTouch:setTouchEnabled(true)
    self._panelTouch:addClickEventListenerEx(function()
        -- body
        if self._rewardCallback then
            self._rewardCallback(curDay)
        end
    end)
end



return CustomActivityWeekFundsV2Cell

File: study/ts/script/scene/view/customActivity/CustomActivityWeekFundsV2Cell.ts
import CommonListItem from "../../../ui/component/CommonListItem";
import EffectGfxNode from "../../../effect/EffectGfxNode";
import CommonUI from "../../../ui/component/CommonUI";
import { Path } from "../../../utils/Path";
import { CustomActivityConst } from "../../../const/CustomActivityConst";
import { Lang } from "../../../lang/Lang";
import { Colors, G_EffectGfxMgr } from "../../../init";

const {ccclass, property} = cc._decorator;

@ccclass
export default class CustomActivityWeekFundsV2Cell extends CommonListItem {

   @property({
       type: cc.Node,
       visible: true
   })
   _resource: cc.Node = null;

   @property({
       type: cc.Sprite,
       visible: true
   })
   _imageNormal: cc.Sprite = null;

   @property({
       type: cc.Sprite,
       visible: true
   })
   _imageShade: cc.Sprite = null;

   @property({
       type: cc.Sprite,
       visible: true
   })
   _imageLarge: cc.Sprite = null;

   @property({
       type: cc.Sprite,
       visible: true
   })
   _imageDay: cc.Sprite = null;

   @property({
       type: cc.Sprite,
       visible: true
   })
   _imageSelected: cc.Sprite = null;

   @property({
       type: cc.Sprite,
       visible: true
   })
   _imageYuanBao: cc.Sprite = null;

   @property({
       type: cc.Label,
       visible: true
   })
   _textNum: cc.Label = null;

   @property({
       type: cc.Node,
       visible: true
   })
   _nodeEffect: cc.Node = null;

   @property({
       type: cc.Label,
       visible: true
   })
   _textState: cc.Label = null;

   @property({
       type: cc.Sprite,
       visible: true
   })
   _imageDuiGou: cc.Sprite = null;

   @property({
       type: cc.Node,
       visible: true
   })
   _panelTouch: cc.Node = null;
   
   private _rewardCallback:any
   setInitData(rewardCallback) {
    this._rewardCallback = rewardCallback;
}
onCreate() {
    var size = this._resource.getContentSize();
    this.node.setContentSize(size.width, size.height);
}
_createEffect(effect, state) {
    if (effect == 0) {
        return;
    }
    var selectedFlash = this['_nodeEffect'].getChildByName('flash_effect');
    if (selectedFlash == null) {
        var lightEffect = G_EffectGfxMgr.createPlayGfx(this['_nodeEffect'],"effect_chongzhi_guangyun6");
        lightEffect.node.setAnchorPoint(0, 0);
        lightEffect.play();
        lightEffect.node.active = (state == 0);
        lightEffect.node.name = ('flash_effect');
        lightEffect.node.setPosition(this['_nodeEffect'].getContentSize().width * 0.5, this['_nodeEffect'].getContentSize().height * 0.5 + 8);
    } else {
        selectedFlash.active = (state == 0);
    }
}
updateUI(data) {
    var effect = parseInt(data.effects);
    var curDay = parseInt(data.day);
    this._imageDay.node.addComponent(CommonUI).loadTexture(Path.getCustomActivityUI(CustomActivityConst.FUNDS_V2_DAY[curDay-1]));
    // this._imageDay.ignoreContentAdaptWithSize(true);
    this._imageYuanBao.node.addComponent(CommonUI).loadTexture(Path.getCustomActivityUI(CustomActivityConst.FUNDS_YUANBAO[effect]));
    // this._imageYuanBao.ignoreContentAdaptWithSize(true);
    this._textNum.string = ('x' + (data.reward_size_1));
    if (data.isActived && data.canSignedDay) {
        this._createEffect(effect, data.canGet);
        this._imageShade.node.active = (data.canGet == 1);
        this._imageDuiGou.node.active = (data.canGet == 1);
        this._imageLarge.node.active = (data.canGet == 0);
        this._imageSelected.node.active = (data.canGet == 0);
        this._panelTouch.active = (data.canGet == 0);
        this._textState.string = (data.canGet == 1 && Lang.get('weekfunds_v2_state2') || Lang.get('weekfunds_v2_state3'));
        this._textState.node.color = (data.canGet == 0 && Colors.FUNDSWEEK_V2_GOT || Colors.FUNDSWEEK_V2_NOTGOT);
    } else {
        this._createEffect(effect, 0);
        this._imageShade.node.active = (false);
        this._imageDuiGou.node.active = (false);
        this._imageLarge.node.active = (false);
        this._imageSelected.node.active = (false);
        this._panelTouch.active = (false);
        this._textState.string = (Lang.get('weekfunds_v2_state1'));
        this._textState.node.color = (Colors.FUNDSWEEK_V2_NOTGOT);
    }

    var btn = this._panelTouch.getComponent(cc.Button) as cc.Button;
    btn.clickEvents = [];
    var newE = new cc.Component.EventHandler();
    newE.component = "CustomActivityWeekFundsV2Cell";
    newE.target = this.node;
    newE.handler = "onCallBack";
    newE.customEventData = curDay.toString();
    btn.clickEvents.push(newE);
}
private onCallBack(sender,day:number){
      if(this._rewardCallback)
      this._rewardCallback(day);
}

}
